### PR TITLE
Fix ApproachRate not inheriting OverallDifficulty when it is undefined

### LINKED ISSUE #2047: ApproachRate should inherit OverallDifficulty if beatmap does not specify it
[Shinjitsu no Tsubasa](https://osu.ppy.sh/s/8492)
This beatmap's osu format version is v5 and it does not have ApproachRate settings.
osu! treat ApproachRate as same of OverallDifficulty:

osu!lazer defaults ApproachRate to [DEFAULT_DIFFICULTY](https://github.com/ppy/osu/blob/master/osu.Game/Beatmaps/BeatmapDifficulty.cs).
This has to be fixed.

### PR DESCRIPTION
**Pull request by sunghwan2789**
Fix #2047

### CODE REVIEW

File: osu.Game/Beatmaps/BeatmapDifficulty.cs [hunk 1]
```diff
@@ -20,7 +20,18 @@ public class BeatmapDifficulty
...
+        {
+            get
+            {
+                return approachRate ?? OverallDifficulty;
+            }
+            set
+            {
+                approachRate = value;
+            }
+        }
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/BeatmapDifficulty.cs]: Add newline.

File: osu.Game/Beatmaps/BeatmapDifficulty.cs [hunk 2]
```diff
@@ -20,7 +20,18 @@ public class BeatmapDifficulty
         public float DrainRate { get; set; } = DEFAULT_DIFFICULTY;
         public float CircleSize { get; set; } = DEFAULT_DIFFICULTY;
         public float OverallDifficulty { get; set; } = DEFAULT_DIFFICULTY;
-        public float ApproachRate { get; set; } = DEFAULT_DIFFICULTY;
```
Comment (smoogipoo) [inline osu.Game/Beatmaps/BeatmapDifficulty.cs]: Add newline.

File: osu.Game/Beatmaps/BeatmapDifficulty.cs (hunk 1 above)
Comment (smoogipoo) [inline osu.Game/Beatmaps/BeatmapDifficulty.cs]: Let's simplify this stuff into:
```
ApproachRate
{
    get => approachRate ?? OverallDifficulty;
    set => approachRate = value;
}
```

**Review by smoogipoo — DISMISSED** [review]

**Comment by peppy** [comment]
I'm just gonna apply these changes so we can get them in today's build.

**Review by peppy — APPROVED** [review]

**Pull request merged**